tests/testing_harness.py

```diff
@@ -271,42 +274,123 @@ class PlottingTestHarness(TestHarness):
+        """Overwrite the reference images with the test images."""
+
+        # Find all plot files
+        outputs = glob.glob(os.path.join(os.getcwd(), 'test-*.png'))
+
+        # Copy each test plot as a new reference plot
+        for i in range(len(outputs)):
+            shutil.copyfile('test-{0}.png'.format(i), 'true-{0}.png'.format(i))
+
+    def _compare_results(self, max_distance=1.):
+        """Make sure the current results agree with the true standard."""
+
+        # Loop over each Matplotlib figure / PIL Image and
+        # compare to reference using Matplotlib fuzzy comparison
+        for i, fig in enumerate(self.figures):
+
+            # Open test image and resize to that of the true image with PIL
+            img1 = Image.open('test-{0}.png'.format(i))
+            img2 = Image.open('true-{0}.png'.format(i))
+            img1 = ImageOps.fit(img1, img2.size, Image.ANTIALIAS)
+
+            # Compute distance between each image in RGB space
+            distance = self._compare_images(img1, img2)
+            assert distance < max_distance, 'Results do not agree.'
 
     def _cleanup(self):
         """Delete plot PNG files."""
 
-        # Find all plot files
-        outputs = glob.glob(os.path.join(os.getcwd(), '*.png'))
+        # Find all test plot files
+        outputs = glob.glob(os.path.join(os.getcwd(), 'test-*.png'))
 
         # Remove each plot file if it exists
-        for output in outputs:
+        for i in range(len(outputs)):
+            output = 'test-{0}.png'.format(i)
             if os.path.isfile(output):
                 os.remove(output)
             elif os.path.isdir(output):
                 shutil.rmtree(output)
 
         super(PlottingTestHarness, self)._cleanup()
+
+    def _compare_images(self, img1, img2):
+        """Compare two PIL images using in RGB space with pixel histograms."""
+
+        # Extract RGBA data from each PIL Image
+        rgba1 = np.array(img1)
+        rgba2 = np.array(img2)
+
+        # Compute histograms of each images pixels
+        hr1, bins1 = np.histogram(rgba1[...,0], bins=256, normed=True)
+        hg1, bins1 = np.histogram(rgba1[...,1], bins=256, normed=True)
+        hb1, bins1 = np.histogram(rgba1[...,2], bins=256, normed=True)
+        hr2, bins2 = np.histogram(rgba2[...,0], bins=256, normed=True)
+        hg2, bins2 = np.histogram(rgba2[...,1], bins=256, normed=True)
+        hb2, bins2 = np.histogram(rgba2[...,2], bins=256, normed=True)
+        hist1 = np.array([hr1, hg1, hb1]).ravel()
+        hist2 = np.array([hr2, hg2, hb2]).ravel()
+        
+        # Compute cartesian distance between histograms in RGB space
+        diff = hist1 - hist2
+        distance = np.sqrt(np.dot(diff, diff))
+        return distance
+
+
+class MultiSimTestHarness(TestHarness):
+    """Specialized TestHarness for testing multi-simulation capabilities."""
+
+    def __init__(self):
+        super(MultiSimTestHarness, self).__init__()
+        self.num_simulations = 3
```

Comment:
Yes you have a good point and I'm open to changing it to 2. The reason I set it to 3 is that many of the PRs I've submitted the last year to enable multiple simulations with a single `Solver`+`TrackGenerator`+`Generator` fixed memory allocation/deallocation issues that randomly broke the code. Similar to some of the issues you faced when in #270, the code would randomly segmentation fault after 2+ simulations rather than always on the second simulation. Since I was burned before, I set this to 3, though I recognize that my argument for doing so would actually indicate that there is no reason to believe that 3 is actually enough. It doesn't dramatically impact the runtime of the test suite as it stands, so I'd prefer to keep it at 3, but am willing to drop to 2 if you strongly prefer it.